app/parsers/bulkrax/oai_dc_parser.rb

```diff
@@ -76,42 +74,38 @@ def list_sets
       client.list_sets
     end
 
-    def run
-      create_collections
-      create_works
-    end
-
     def create_collections
+      metadata = {
+        visibility: 'open',
+        collection_type_gid: Hyrax::CollectionType.find_or_create_default_collection_type.gid
+      }
+
       list_sets.each do |set|
-        if collection_name == 'all' || collection_name == set.spec
-          attrs = {
-            title: [set.name],
-            identifier: [set.spec],
-            contributing_institution: [parser_fields['institution_name']],
-            visibility: 'open',
-            collection_type_gid: Hyrax::CollectionType.find_or_create_default_collection_type.gid
-          }
-          #Bulkrax::CollectionFactory.new(attrs).find_or_create
-          collection = Collection.where(identifier: [set.spec]).first
-          collection ||= Collection.create!(attrs)
+        next unless collection_name == 'all' || collection_name == set.spec
+
+        metadata[:title] = [set.name]
+        metadata[Bulkrax.system_identifier_field] = [set.spec]
+
+        new_entry = collection_entry_class.where(importer: importer, identifier: set.spec, raw_metadata: metadata).first_or_create!
+        acquire_lock_for(importer.current_importer_run.id) do
+          ImportWorkCollectionJob.perform_later(new_entry.id, importer.current_importer_run.id)
```

Comment:
same issue in that you don't know when this gets finished queue wise as mentioned above